pkg/generic/binarythrift_codec.go

```diff
@@ -51,10 +51,8 @@ func (c *binaryThriftCodec) Marshal(ctx context.Context, msg remote.Message, out
 	var transBuff []byte
 	var ok bool
 	if msg.RPCRole() == remote.Server {
-		gResult := data.(*Result)
-		transBinary := gResult.Success
 		// handle biz error
-		if transBinary == nil {
+		if msg.RPCInfo().Invocation().BizStatusErr() != nil {
```

Comment:
Business error only work properly when using TTHeader and HTTP2 transmission protocols. If there is a business error, `data.(*Result).Success` will be nil, and an empty payload will be constructed here to return

Reply:
please add this comment to clarify the reason for the code logic here